phoenix-core/src/main/java/org/apache/phoenix/schema/stats/DefaultStatisticsCollector.java

```diff
@@ -220,68 +186,62 @@ public void close() throws IOException {
-                    }
-                    if (!guidePostsInfoWriterMap.containsKey(cfKey)) {
-                        Pair<Long, GuidePostsInfoBuilder> emptyGps = new Pair<Long, GuidePostsInfoBuilder>(0l, new GuidePostsInfoBuilder());
-                        guidePostsInfoWriterMap.put(cfKey, emptyGps);
-                    }
+        Set<ImmutableBytesPtr> fams = guidePostsInfoWriterMap.keySet();
+        // Update the statistics table.
+        // Delete statistics for a region if no guide posts are collected for that region during
+        // UPDATE STATISTICS. This will not impact a stats collection of single column family during
+        // compaction as guidePostsInfoWriterMap cannot be empty in this case.
+        if (cachedGuidePosts == null) {
+            // We're either collecting stats for the data table or the local index table, but not both
+            // We can determine this based on the column families in the scan being prefixed with the
+            // local index column family prefix. We always explicitly specify the local index column
+            // families when we're collecting stats for a local index.
+            boolean collectingForLocalIndex = scan != null &&
+                    !scan.getFamilyMap().isEmpty() &&
+                    MetaDataUtil.isLocalIndexFamily(scan.getFamilyMap().keySet().iterator().next());
+            for (Store store : region.getStores()) {
+                ImmutableBytesPtr cfKey = new ImmutableBytesPtr(store.getFamily().getName());
+                boolean isLocalIndexStore = MetaDataUtil.isLocalIndexFamily(cfKey);
+                if (isLocalIndexStore != collectingForLocalIndex) {
+                    continue;
                 }
-            }
-            for (ImmutableBytesPtr fam : fams) {
-                if (delete) {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Deleting the stats for the region " + region.getRegionInfo());
-                    }
-                    statsWriter.deleteStatsForRegion(region, this, fam, mutations);
-                }
-                if (logger.isDebugEnabled()) {
-                    logger.debug("Adding new stats for the region " + region.getRegionInfo());
-                }
-                // If we've disabled stats, don't write any, just delete them
-                if (this.guidePostDepth > 0) {
-                    statsWriter.addStats(this, fam, mutations);
+                if (!guidePostsInfoWriterMap.containsKey(cfKey)) {
+                    Pair<Long, GuidePostsInfoBuilder> emptyGps = new Pair<Long, GuidePostsInfoBuilder>(0l, new GuidePostsInfoBuilder());
+                    guidePostsInfoWriterMap.put(cfKey, emptyGps);
                 }
             }
-        } catch (IOException e) {
-            logger.error("Failed to update statistics table!", e);
-            throw e;
+        }
+        for (ImmutableBytesPtr fam : fams) {
+            if (delete) {
```

Comment:
from line 228 to line 237, you removed the conditional debug logs and add info log, not very sure this is better and necessary.

Reply:
maybe still use debug log is better.

Reply:
All the log lines were changed to INFO since they will be produced only once per mapper (and a mapper would correspond to a region). These lines would really help us in debugging if something goes wrong. We can also export these as metrics if required.